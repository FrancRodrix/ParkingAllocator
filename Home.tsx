import {
  StyleSheet,
  Text,
  View,
  TextInput,
  SafeAreaView,
  TouchableOpacity,
  KeyboardAvoidingView,
} from 'react-native';
import React, {useState, useEffect} from 'react';


interface Props {
  navigation: any;
  route: any;
}

export default function Home(props:Props) {
  const [slots, setSlots] = useState('');

  return (
    <KeyboardAvoidingView style={styles.container}>
      <Text style={styles.head}> Parking Allocator</Text>
      <TextInput
        // onChangeText={(text: any) => {
        //   setSlots(text);
        // }}
     
        value={slots}
        style={styles.textInput}
        placeholder="SLOTS TO BOOK"
        placeholderTextColor={'grey'}
        keyboardType={'numeric'}
      />
      <TouchableOpacity
        style={styles.submitButton}
        onPress={() => {
          if(slots){
            setSlots('')
            props.navigation.navigate('ParkScreen', {
               slots: slots,
             })
          }       
          
        }}>
        <Text style={styles.submitText}>SUBMIT</Text>
      </TouchableOpacity>
      </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#f0e130',
  },
  head: {
    fontSize: 80,
    fontWeight: '100',
    alignSelf: 'center',
    // position: 'absolute',
    color: 'white',
    marginTop: -200,
    // top: 80,
  },
  textInput: {
    // borderWidth:1,
    borderColor: '#000',
    borderRadius: 10,
    width: '70%',
    textAlign: 'center',
    letterSpacing: 5,
    backgroundColor: '#fff',
  },
  submitButton: {
    width: '70%',
    backgroundColor: '#DE3163',
    borderRadius: 10,
    marginTop: 15,
    padding: 15,
  },
  submitText: {
    color: '#fff',
    letterSpacing: 10,
    fontWeight: '700',
    textAlign: 'center',
  },
});
