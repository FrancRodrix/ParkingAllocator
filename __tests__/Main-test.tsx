import 'react-native'
import React from 'react'
import Home from '../Class/Screens/Main'
import renderer from 'react-test-renderer'
import {render, screen, fireEvent} from '@testing-library/react-native'



const navigation = {
    navigate: jest.fn()  
    // params:null,
  };

test('Home snapshot',()=>{
    const snap=renderer.create(
        <Home/>
    ).toJSON();
    expect(snap).toMatchSnapshot();
});

let findElement=function(tree:any,element:any){
    return true
}

it('find element',()=>{
    let tree=renderer.create(
        <Home  navigation={navigation} />
    ).toJSON();
    expect(findElement(tree,'inputData')).toBeDefined();


})

test('examples of some things', async () => {
    const expectedNumber = 4
    render(<Home/>)
    const tree = renderer.create(<Home navigation={navigation} />);

    fireEvent.changeText(screen.getByTestId('inputData'))
    fireEvent.press(screen.getByTestId('SubmitButton'))
    const textInput = tree.root.findByProps({ testID: "inputData" }).props;
    renderer.act(() => textInput.onChangeText("Test"));
    const textInput1 = tree.root.findByProps({ testID: "inputData" }).props;
    expect(textInput1.value).toBe("Test");
    expect(expectedNumber).toBe(4)
    expect(screen.toJSON()).toMatchSnapshot()
})

// test("Navigate correctly", () => {
//     const tree = renderer.create(<Home navigation={navigation} />);
//     const button = tree.root.findByProps({ testID:"SubmitButton" }).props;
//     button.onPress();
//     expect(navigation.navigate).toBeCalledWith("Detail", {
//       slots:'',
//     });
//   });





