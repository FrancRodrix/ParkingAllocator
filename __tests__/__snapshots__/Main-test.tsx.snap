// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f0e130",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 80,
        "fontWeight": "100",
        "marginTop": -200,
      }
    }
  >
     Parking Allocator
  </Text>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="SLOTS TO BOOK"
    placeholderTextColor="grey"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderRadius": 10,
        "letterSpacing": 5,
        "textAlign": "center",
        "width": "70%",
      }
    }
    testID="inputData"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#DE3163",
        "borderRadius": 10,
        "marginTop": 15,
        "opacity": 1,
        "padding": 15,
        "width": "70%",
      }
    }
    testID="SubmitButton"
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontWeight": "700",
          "letterSpacing": 10,
          "textAlign": "center",
        }
      }
    >
      SUBMIT
    </Text>
  </View>
</View>
`;

exports[`examples of some things 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f0e130",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 80,
        "fontWeight": "100",
        "marginTop": -200,
      }
    }
  >
     Parking Allocator
  </Text>
  <TextInput
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="SLOTS TO BOOK"
    placeholderTextColor="grey"
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderRadius": 10,
        "letterSpacing": 5,
        "textAlign": "center",
        "width": "70%",
      }
    }
    testID="inputData"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#DE3163",
        "borderRadius": 10,
        "marginTop": 15,
        "opacity": 1,
        "padding": 15,
        "width": "70%",
      }
    }
    testID="SubmitButton"
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontWeight": "700",
          "letterSpacing": 10,
          "textAlign": "center",
        }
      }
    >
      SUBMIT
    </Text>
  </View>
</View>
`;
